### ISSUE TITLE
Tasks for CAPI release v1.1

### ISSUE THREAD

**Issue by sbueringer** [kubernetes-sigs/cluster-api#5967]
### During the release cycle

* [ ] GitHub milestone
    * Create the new milestone
    * [TBD] Move issues
    * Update the [milestone_applier config](https://github.com/kubernetes/test-infra/blob/0b17ef5ffd6c7aa7d8ca1372d837acfb85f7bec6/config/prow/plugins.yaml#L371) accordingly
* [ ] Update top-level metadata.yaml
* [ ] Book:
    * Add a new migration doc for provider implementers
* [ ] Check if we should remove previously deprecated code
* [ ] Update docker.yaml: update `providers:`
* [ ] Make sure clusterctl is able to upgrade to the new version
    * Mainly just test it to make sure that things like `CheckCAPIContract` work as expected 

* [ ] If we upgrade the apiVersion
    * Add new version of the types:
        * Create new api packages by copying existing packages
        * Make sure webhooks only exist in the latest apiVersion (same for other subpackages like `index`)
        * Add conversion and conversion tests
        * Adjust generate targets in the Makefile
    * Update import aliases in .golangci.yml
    * Switch other code over to the new version (imports across the code base, e.g. controllers)
        * Add all versions to the schema in the `main.go`'s
    * Add types to the `PROJECT` files of the respective provider
    * Add test data for the new version in test/e2e/data/{infrastructure-docker,shared} (also update top-level `.gitignore`)
    * Update docker.yaml, make sure all tests are successful in CI

* [ ] Make sure we're using the latest Kubernetes / controller-runtime version available (see Kubernetes housekeeping tasks)

### When releasing

* [ ] Create a release branch release-x.x
* [ ] Create jobs and dashboard (config/testgrids/kubernetes/sig-cluster-lifecycle/config.yaml) in test-infra for the new release branch
    * *Note* don't forget clusterctl upgrade jobs
* [ ] If we upgraded the apiVersion:
    * Book:
        * Make sure the book is available under `<prefix>.cluster-api.sigs.k8s.io` (URL TBD)
        * Link the "old" book on the start page of the current book (main branch)

* [ ] The actual [release](https://github.com/kubernetes-sigs/cluster-api/blob/main/docs/developer/releasing.md)

### Before the next cycle starts

* [ ] Duplicate this issue to track the next release

**Comment by sbueringer** [comment]
Release branch has been created, now we can create the jobs and the dashboard. I've created a separated issue for that: https://github.com/kubernetes-sigs/cluster-api/issues/5970 (help wanted)

**Comment by alexander-demichev** [comment]
> Modify the clusterctl upgrade jobs for testing the new release
does this mean adding a job for upgrading v1.1 to main?

**Comment by sbueringer** [comment]
> > Modify the clusterctl upgrade jobs for testing the new release
> 
> does this mean adding a job for upgrading v1.1 to main?

Good question. v1.1 had/has:
* clusterctl v0-3 to main
* clusterctl v1-0 to main
* capi-e2e-main tests v0-4 to main (as the docker.yaml on main/v1.1 is configured to upgrade from v0.4)

I think for v1.2 we only have to adjust the `clusterctl v1-0 to main` job to upgrade from v1-1 to main.

The idea is to then have jobs which test from the latest release of each currently supported contract version (v1alpha3 (v0.3.x), v1alpha4 (v0.4.x), v1beta1 (v1.1.x). I think we have to modify the docker.yaml first accordingly and wait until v1.1 has been released (current plan is 2nd February afaik)

**Comment by sbueringer** [comment]
@fabriziopandini Please correct me if I'm wrong.

**Comment by fabriziopandini** [comment]
I agree, what is important is to test upgrades where something relevant changes (e.g API version) + upgrade from the latest version, so we need coverage for
- v0-3 to main
- v0-4 to main
- v1-1 to main (that should replace clusterctl v1-0 to main)

**Comment by alexander-demichev** [comment]
I guess we'll also need to bump versions here https://github.com/kubernetes-sigs/cluster-api/blob/main/cmd/clusterctl/hack/create-local-repository.py#L56?

**Comment by sbueringer** [comment]
> I guess we'll also need to bump versions here https://github.com/kubernetes-sigs/cluster-api/blob/main/cmd/clusterctl/hack/create-local-repository.py#L56?

Yes. Essentially you have to grep for ~ `v1.1` and take a look what we have to change.

**Comment by sbueringer** [comment]
Short update. We're making good progress, thx @alexander-demichev!

Currently still open are:
* #6103: good first issue, for new contributors
* #6017: requires discussion with Netlify

I would close the current issue once #6103 is done. The book is basically in a good state, #6017 is a improvement which seems to take a bit more time.

**Comment by sbueringer** [comment]
I'll close this issue for release 1.1=>1.2 everything is done 
(the CAPI book versioning issue stays open for a bit for some backports to older books)

Thx to everyone who contributed!

I tried to keep the list as complete as possible, so it's easy to repeat for the next release. Please let me know if I missed something.

/close

**Comment by sbueringer** [comment]
@fabriziopandini Should we open the new issue now so we don't forget it or right before the v1.2 release?

**Comment by fabriziopandini** [comment]
Let's wait until we have an idea about the v1.2 release date

BTW, this is great work! I'm sure we will greatly benefit from this!
